db/errors.go

```diff
@@ -0,0 +1,6 @@
+package db
+
+import mgo "gopkg.in/mgo.v2"
+
+func IsDuplicateKey(err error) bool  { return mgo.IsDup(err) }
+func ResultsNotFound(err error) bool { return err == mgo.ErrNotFound }
```

Comment:
I don't think this works since errors are pointers to a string. You'll need to check err.Error()